src/pages/repository_settings_secret/content.js

```diff
@@ -0,0 +1,212 @@
+                          eventTypes.map(eventType => {
+                            return (
+                              <Switch key={eventType.value} id={eventType.value} disabled={true} checked={ secret.event.indexOf(eventType.value) !== -1}>{eventType.label}</Switch>
+                            );
+                          })
+                        }
+                      </Cell>
+                      <Cell col={6}>
+                        <Checkbox label="Skip Verify" disabled={true} checked={ secret.skip_verify }/>
+                        <Checkbox label="Conceal" disabled={true} checked={ secret.conceal}/>
+                      </Cell>
+                    </Grid>
+                  </div>
+                  <hr/>
+                </div>
+              );
+            })
+        }
+
+        <Textfield
+          label="Secret Name"
+          floatingLabel
+          style={{ width: '300px' }}
+          onChange={(field) => this.handleAddKeyChange(field.target.value)}
+          value={this.state.addSecret.name}
+        />
+        <Textfield
+          label="Secret Value"
+          floatingLabel
+          style={{ width: '300px' }}
+          onChange={(field) => this.handleAddSecretChange(field.target.value)}
+          value={this.state.addSecret.value}
+        />
+        <FABButton ripple mini
+                   onClick={this.handleAdd.bind(this, owner, name)}
+                   disabled={ this.state.addSecret.name === '' || this.state.addSecret.value === ''}
+        >
+          <Icon name="add"/>
+        </FABButton>
+        <div style={{ width: '100%', margin: 'auto' }}>
+          <Grid className="secret-properties">
+            <Cell col={6}>
+              {
+                eventTypes.map(eventType => {
+                  return (
+                    <Switch key={eventType.value} id={eventType.value} onChange={ (field) => this.handleAddEventSwitch(field) }
+                            checked={ this.state.addSecret.event.indexOf(eventType.value) !== -1}>{eventType.label}</Switch>
+                  );
+                })
+              }
+            </Cell>
+            <Cell col={6}>
+              <Checkbox id="skip_verify" label="Skip Verify" onChange={(field) => this.handleOptionalChange(field)}
+                        checked={ this.state.addSecret.skip_verify }/>
+              <Checkbox id="conceal" label="Conceal" onChange={(field) => this.handleOptionalChange(field)}
+                        checked={ this.state.addSecret.conceal }/>
+            </Cell>
+          </Grid>
+        </div>
+
+      </PageContent>
+    );
+  }
+
+  handleDelete(owner, name, secret) {
+    events.emit(DEL_REPO_SECRET, { owner, name, secret });
+  }
+
+  handleUpdateSecret(owner, name, secretKey) {
+    //TODO: need to handle skip_verify and what events
+    let secret = {
+      name: secretKey,
+      value: this.state[secretKey],
+      event: ['push', 'tag', 'deployment'],
+      image: ['*'],
```

Comment:
we should remove `*` because, until the old code is removed, this instructs drone to inject the secret into every container.